bundles/binding/org.openhab.binding.ecobee/src/main/java/org/openhab/binding/ecobee/internal/messages/Temperature.java

```diff
@@ -132,7 +133,7 @@ public static Temperature fromFahrenheit(BigDecimal fahrenheit) {
 	 *            the Celsius temperature
 	 */
 	public static Temperature fromCelsius(BigDecimal celsius) {
-		return new Temperature(celsius.multiply(NINE).divide(FIVE).add(THIRTY_TWO).movePointRight(1));
+		return new Temperature(celsius.multiply(NINE).divide(FIVE, MathContext.DECIMAL32).add(THIRTY_TWO).movePointRight(1));
```

Comment:
That would make more sense.  I used some BigDecimal F-to-C example from somewhere that does it this way, and most conversion examples in a Google search seem to prefer 9/5 (probably because they are explaining how to convert in your head).  A single multiply would virtually certainly be faster, since there is one fewer intermediate objects created.  But since I have a user who has already signed off on the current code, I will leave it unless Thomas would like me to do it differently.  Thanks, Rob, for the comment!